web/src/compose_setup.js

```diff
@@ -523,11 +523,18 @@ export function initialize() {
     });
 
     $("input#stream_message_recipient_topic").on("focus", () => {
-        compose_recipient.update_compose_area_placeholder_text();
+        const $input = $("input#stream_message_recipient_topic");
+        compose_recipient.update_topic_displayed_text($input.val(), true);
+        compose_recipient.update_compose_area_placeholder_text(false);
```

Comment:
I'm pondering whether `update_topic_displayed_text()` ought to call `update_compose_area_placeholder_text()`, since the former has better logic for handling _general chat_ and needs to keep the compose area's placeholder text in sync, anyway.